site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,970 @@
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+----
+
+The two FMP goals that we specify are `resource` and `build`.
+
+link:https://maven.fabric8.io/#fabric8:resource[resource] is automatically attached to the `process-resources` phase of the default Maven lifecycle
+and it's purpose is to create the OpenShift resources (as files on the file system) that are needed to deploy the application. +
+These resources include the `Service`, `DeploymentConfig` and `Route` which are all necessary to make the application available outside the cluster. +
+
+It's worth noting that the produced OpenShift resources are placed inside `target/classes/META-INF/fabric8` and the specific file that will be used later
+by Arquillian is `target/classes/META-INF/fabric8/openshift.yml`. +
+Having the files under `target/classes` means that when Maven packages the application and creates the final artifact (the Spring Boot uber-jar),
+the all too important `META-INF/fabric8/openshift.yml` file will be present inside it.
+
+link:https://maven.fabric8.io/#fabric8:build[build] is automatically attached to the `pre-integration-test` phase of the default Maven lifecycle
+and takes care of applying the `BuildConfiguration` and `ImageStream` resources to the cluster that are necessary to deploy the application later on.
+
+[NOTE]
+====
+A great way to see the plugins that are attached to each Maven lifecycle phase is by executing the following command:
+[source,bash]
+----
+mvn fr.jcgay.maven.plugins:buildplan-maven-plugin:list-phase -Popenshift,openshift-it
+----
+====
+
+
+==== How FMP resources are used before the tests are run
+
+Now that we have a firm grasp on how and when FMP creates the necessary OpenShift resources, we will close the loop and describe how these resources are used
+in order to launch the application before Arquillian starts the actual tests.
+
+As should be evident from the previous section, by the time Arquillian launches (as part of the `integration-test` phase of the default Maven lifecycle),
+FMP has taken care of two things:
+
+* An `ImageStream` has been applied on OpenShift that contains the Docker image built from the application's uber-jar
+* A YAML file containing all the OpenShift resources necessary to launch the application has been created at `target/classes/META-INF/fabric8/openshift.yml`
+
+The aforementioned `openshift.yml` file has also been copied to the Spring Boot uber-jar (with the `META-INF` directory structure preserved).
+This is important since as was described earlier in the guide, the uber-jar is part of the test classpath created by the failsafe plugin.
+
+When Arquillian is launched, if it has been configured to have `env.init.enabled` set to `true`, it dutifully looks for `META-INF/fabric8/openshift.yml`
+inside it's classpath. Once the file is found, it used to deploy the application to OpenShift.
```

Comment:
it's -> its

Reply:
Keeps on happening... It's a good thing I am not a writer... my editor would have a horrible time correcting this kind of thing :P